mathics/builtin/linalg.py

```diff
@@ -693,42 +699,81 @@ def apply(self, m, evaluation):
 class Eigenvalues(Builtin):
     """
     <dl>
-    <dt>'Eigenvalues[$m$]'
-        <dd>computes the eigenvalues of the matrix $m$.
+      <dt>'Eigenvalues[$m$]'
+      <dd>computes the eigenvalues of the matrix $m$.
+      By default Sympy's routine is used. Sometiems this is slow and
+      less good than the corresponding mpmath routine. Use option Method->"mpmath" if you want
+      to use mpmath's routine instead.
     </dl>
 
-    >> Eigenvalues[{{1, 1, 0}, {1, 0, 1}, {0, 1, 1}}] // Sort
-     = {-1, 1, 2}
+    Numeric eigenvalues are sorted in order of decreasing absolute value:
+    >> Eigenvalues[{{1, 1, 0}, {1, 0, 1}, {0, 1, 1}}]
+     = {2, -1, 1}
 
+    Symbolic eigenvalues:
     >> Eigenvalues[{{Cos[theta],Sin[theta],0},{-Sin[theta],Cos[theta],0},{0,0,1}}] // Sort
      = {1, Cos[theta] + Sqrt[(-1 + Cos[theta]) (1 + Cos[theta])], Cos[theta] - Sqrt[(-1 + Cos[theta]) (1 + Cos[theta])]}
 
     >> Eigenvalues[{{7, 1}, {-4, 3}}]
      = {5, 5}
 
+    >> Eigenvalues[{{7, 1}, {-4, 3}}]
+     = {5, 5}
+
     #> Eigenvalues[{{1, 0}, {0}}]
      : Argument {{1, 0}, {0}} at position 1 is not a non-empty rectangular matrix.
      = Eigenvalues[{{1, 0}, {0}}]
+
+    Compute Eigenvalues using mpmath's routines; Sympy is slow here and returns
+    complex numbers.
+    >> Eigenvalues[{{-8, 12, 4}, {12, -20, 0}, {4, 0, -2}}, Method->"mpmath"]
+     = {{0.842134, 0.396577, 0.365428}, {-0.5328, 0.507232, 0.677377}, {0.0832756, -0.765142, 0.638454}}
     """
 
+    sympy_name = "eigenvalues"
+    mpmath_name = "eig"
+
     messages = {
         'matrix': "Argument `1` at position `2` is not a non-empty rectangular matrix.",
     }
 
-    def apply(self, m, evaluation):
-        'Eigenvalues[m_]'
+    @staticmethod
+    def mp_eig(mp_matrix) -> Expression:
+        try:
+            _, ER = mp.eig(mp_matrix)
+        except:
+            return None
 
-        matrix = to_sympy_matrix(m)
-        if matrix is None:
+        eigenvalues = ER.tolist()
+        # Sort the eigenvalues in the Mathematica convention: largest first.
+        eigenvalues.sort(key=lambda v: (abs(v[0]),-v[0].real,-(v[0].imag)), reverse=True)
```

Comment:
We should probably extract this (the sorting routine) to a separate function to avoid duplication of code (so that we don't forget to update both lines if we want to update this in the future). This should be inside of a `try` block too.